File: src/main/java/com/hyungu/whoseturn/entities/RoomUserEntity.java
package com.hyungu.whoseturn.entities;

import jakarta.persistence.*;

@Entity
@IdClass(RoomUserEntity.class)
public class RoomUserEntity {

    @Id
    @ManyToOne
    @JoinColumn(name = "ROOM_ID")
    private RoomEntity room;

    @Id
    @ManyToOne
    @JoinColumn(name = "USER_ID")
    private UserEntity user;

    @Column(name = "NAME", nullable = true)
    private String roomName;

}

File: src/main/kotlin/com/hyungu/whoseturn/entities/RoomUserEntity.kt
package com.hyungu.whoseturn.entities

import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.Table

@Table(name = "room_user")
data class RoomUserEntity(
    @Id
    @JoinColumn(name = "id", table = "rooms")
    val roomId: String,

    @Id
    @JoinColumn(name = "id", table = "users")
    val userId: String,

    var name: String
)